.github/workflows/build.yaml

```diff
@@ -156,6 +158,24 @@ jobs:
         env:
           GHA_REF: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.ref || github.ref }}
 
+      - uses: actions-ecosystem/action-regex-match@v2
+        id: match-tag
+        with:
+          text: ${{ github.event.workflow_run.head_branch }}
+          regex: '^v([0-9]+\.\d+\.\d+)$'
+
+      - name: Codesign and notarize binaries
+        if: ${{ steps.match-tag.outputs.match != '' }}
```

Comment:
Is this tag matching only to make sure that this is not executed for `master` builds and pull requests? If yes, then we don't we want to execute this step for `master` builds and pull requests?

Reply:
oh, completely missed that. will update the pattern to work with master branch as well run the workflow for tag push.

thanks for pointing this out!

Reply:
If we want to run it also on `master`, then do we need the `actions-ecosystem/action-regex-match` step at all?

Reply:
we don't want it to sign binaries which are generated for all the commits made on an open PR. The jobs would currently run for any open PR as well as newly pushed tags and any push to the master branch:

```yaml
on:
  pull_request:
  push:
    branches:
      - master
    tags:
      - "v*"
```
The regex-match action limits codesigning to only work with `main|master` branch or a semver tag